index.html

```diff
@@ -1085,48 +1108,6 @@ <h2><dfn>Predicted events</dfn></h2>
                 </code>
             </pre>
         </section>
-        
-        <section>
-            <h2><dfn>Populating and maintaining the coalesced and predicted event lists</dfn></h2>
-            <p>When a <a>PointerEvent</a> is created, run the following steps for each event in the
-            <a>coalesced event list</a> and <a>predicted event list</a>:</p>
-            <ol>
-                <li>Set the event's <code>pointerId</code>, <code> pointerType</code>,
-                <code>isPrimary</code> and <code>isTrusted</code> to the <var>PointerEvent</var>'s
-                <code>pointerId</code>, <code> pointerType</code>, <code>isPrimary</code> and
-                <code>isTrusted</code>.</li>
-                <li>Set the event's <code>cancelable</code> and <code>bubbles</code> attributes to false.</li>
-                <li>Set the event's <a>coalesced event list</a> and <a>predicted event list</a> to an empty list.</li>
-                <li>Initialize the rest of the attributes the same way as <a>PointerEvent</a>.</li>
-            </ol>
-               
-            <p>A <a>PointerEvent</a> has an associated <dfn>coalesced events targets dirty</dfn> and an associated
-            <dfn>predicted events targets dirty</dfn> flag. When an event is created they must be initialized to false.</p>
-                
-            <p>When the <a>PointerEvent</a>'s target is changed, set <a><code>coalesced events targets dirty</code></a>
-            and <a><code>predicted events targets dirty</code></a> to true.</p>
-
-            <p>When the <code><a data-lt="PointerEvent.getCoalescedEvents">getCoalescedEvents</a></code> method is called:</p>
```

Comment:
How can we remove this? There is now no explanation what happens when getCoalescedEvents() is called.
Same with getPredictedEvents()

Reply:
in the API definition it does say that these two methods return the relevant list https://w3c.github.io/pointerevents/#dom-pointerevent-getcoalescedevents / https://w3c.github.io/pointerevents/#dom-pointerevent-getpredictedevents

as it does seem that the "steps" we had before were just a browser-internal process to normalise the target, and it seems it's irrelevant when the target is actually set (either as part of the `getCoalescedEvents()` / `getPredictedEvents()` calls, or at the time these are gathered/generated initially) as long as the end result is indeed the same `target` as the main/parent/whatever we want to call it pointerevent, it seems  this can be removed?

Reply:
Hmm, the links aren't right in the API.
They should point to ' coalesced event list' and ' predicted event list'
And if *dirty flags aren't used, how do the events in the lists get retargeted when event is patched to shadow DOM?

Reply:
the events will always have the target of the pointer event that the method was called on, if i understand it correctly. so whether the browser does them all in a oner (just goes through the list, sets the target, returns it) or does some internal dirty flag setting to decide if this step is needed or not seems more like a browser-internal optimisation. unless i'm misunderstanding the shadow DOM aspect here?

Reply:
Each time the target of the pointer event changes, also target of the events the get*Events() methods return need to change.
And while event is propagating through shadow + light DOM, the target may change several times.

Reply:
i'm assuming (not knowing about the specific of how it's actually done in-engine) that for each pointer event that is dispatched, the lists are "fresh" and relating only to that specific event, and the target only would always be the target at the exact moment the event's method was called, i.e. (as @flackr asked in the other discussion) as an author you wouldn't be able to observe the retargeting, right? and then if so, it's more an implementation choice when/how the target for all events in the list that is returned when calling the methods is changed to match the target of the event where the method was called?

i mean i'm happy to reinclude the bit that i ripped out here, of course. but it still makes me wonder if it's anything specifically aimed at JS authors, or more of a "how user agents could potentially implement this", or even "one way, but there may be other ways, as long as the end result is that the target of these events in the lists has the same target as the event the method was called on"

Reply:
The flag makes the algorithm rather clear. And in fact, the proposed spec change isn't doing exactly the same as the current spec. Currently if one calls getCoalescedEvents() (just once) and then keeps references to those events and checks their target, .target never changes, but with the proposed change it is unclear whether the target does change at some point if the main event propagates through shadow DOM.

Reply:
but how do you keep a reference to one of the coalesced/predicted events in the list? Aren't they fungible? As soon as a new pointerevent is actually fired, it'd have a whole set of new coalesced/predicted events, no? or are they sort of recycled/kept?